ideascube/mediacenter/views.py

```diff
@@ -10,55 +10,48 @@
 from django.views.generic import (CreateView, DeleteView, DetailView, ListView,
                                   UpdateView)
 
-from ideascube.mixins import ByTagListView
 from ideascube.decorators import staff_member_required
 
 from .models import Document
 from .forms import DocumentForm
 
 
-class KindMixin(object):
-
-    def get_queryset(self):
-        qs = super(KindMixin, self).get_queryset()
-        kind = self.request.GET.get('kind')
-        if kind:
-            qs = qs.filter(kind=kind)
-        return qs
-
-    def get_context_data(self, **kwargs):
-        context = super(KindMixin, self).get_context_data(**kwargs)
-        context['kind'] = self.request.GET.get('kind')
-        return context
-
-
-class Index(KindMixin, ListView):
+class Index(ListView):
     model = Document
     template_name = 'mediacenter/index.html'
     paginate_by = 24
 
     def get_context_data(self, **kwargs):
         context = super(Index, self).get_context_data(**kwargs)
-        context['q'] = self.request.GET.get('q', '')
+        for key in ('q', 'kind', 'lang'):
+            context[key] = self.request.GET.get(key)
+        context['tags'] = self.request.GET.getlist('tags')
+        default_values = {k: context[k] for k in ('kind', 'lang', 'tags')
+                          if context[k]}
+        context['default_values'] = default_values
+        context['not_empty_kinds'] = [
+            (kind, label) for kind, label in Document.KIND_CHOICES
+            if len(Document.objects.filter(kind=kind))
```

Comment:
I think the idea here is to have this dynamic according to current request: so if I'm filtering on, say, tag "plane", I want to see only kinds that have at list one Document with "plane" tag, and so on.
The subtle part is that if we are for example already filtering by some kind, we want to see all other kinds available for the other filters (tags, lang).
So we need to do something like (example for kind, but should be the same with lang):
- get the filters
- remove the "kind" filter if any
- issue a group by on the search table keeping the other filters
- return those kinds

Also, at this point, we certainly want to have a `FilterableView` mixin to factorize this between blog, mediacenter and library.